### PR TITLE
Verify sessions at the frontend

### PR DESCRIPTION
**Pull request by tirr-c**
Session resurrection when the webpage is refreshed.

### CODE REVIEW

File: frontend/src/app/sign-in.component.spec.ts [hunk 1]
```diff
@@ -58,14 +58,13 @@ describe('SignInComponent', () => {
...
         { provide: Router, useValue: routerStub },
-        provideMockActions(observableNever),
       ],
     }).compileComponents()
     .then(() => {
       fixture = TestBed.createComponent(SignInComponent);
       comp = fixture.componentInstance;
       store = fixture.debugElement.injector.get(Store);
-      dispatchSpy = spyOn(store, 'dispatch');
+      dispatchSpy = spyOn(store, 'dispatch').and.callThrough();
```
Comment (skystar-p) [inline frontend/src/app/sign-in.component.spec.ts]: ㅋㅋ

File: frontend/src/app/sign-in.component.ts [hunk 2]
```diff
@@ -19,17 +19,23 @@ export class SignInComponent implements OnInit {
...
+    const user = this.store.select('user');
+    this.verifying = user.map(user => user.verifying);
+    this.error = user.map(user => user.currentError);
   }
 
   ngOnInit() {
-    this.store.dispatch(new UserActions.RedirectIfSignedIn());
+    this.store.dispatch(new UserActions.RedirectWithSignInState({
+      when: 'signed-in',
+      target: 'lobby',
```
Comment (skystar-p) [inline frontend/src/app/sign-in.component.ts]: `target` 대신 `goto`로 바꾸면 완전 자연스럽게 읽힌다..!

**Comment by skystar-p** [comment]
@web-mighty-homu r+

**Comment by web-mighty-homu** [comment]
:pushpin: Commit 468f053 has been approved by `skystar-p`

**Comment by web-mighty-homu** [comment]
:hourglass: Testing commit 468f053881d9ae6edb41b04d80018c0378c6629e with merge d3c2abadcbada46a1411c501e8cef00f5f4d7549...

**Comment by web-mighty-homu** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/swsnu/swpp17-team2/builds/300936378?utm_source=github_status&utm_medium=notification)
Approved by: skystar-p
Pushing d3c2abadcbada46a1411c501e8cef00f5f4d7549 to master...

**Pull request merged**